packages/app-store/_utils/useAddAppMutation.ts

```diff
@@ -28,10 +29,21 @@ type UseAddAppMutationOptions = CustomUseMutationOptions & {
 
 function useAddAppMutation(_type: App["type"] | null, allOptions?: UseAddAppMutationOptions) {
   const { returnTo, ...options } = allOptions || {};
+  const router = useRouter();
+  const onErrorReturnTo = `${WEBAPP_URL}${router.asPath}`;
+
   const mutation = useMutation<
     AddAppMutationData,
     Error,
-    { type?: App["type"]; variant?: string; slug?: string; isOmniInstall?: boolean; teamId?: number } | ""
+    | {
+        type?: App["type"];
+        variant?: string;
+        slug?: string;
+        isOmniInstall?: boolean;
+        teamId?: number;
+        onErrorReturnTo?: string;
+      }
+    | ""
```

Comment:
Lets extract this to a type 
```suggestion
type UseAddAppMutationVariables =
  | {
      type?: App["type"];
      variant?: string;
      slug?: string;
      isOmniInstall?: boolean;
      teamId?: number;
      onErrorReturnTo?: string;
    }
  | undefined;
```